infra/pipeline.ts

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/env node
+import codebuild = require('@aws-cdk/aws-codebuild');
+import codepipeline = require('@aws-cdk/aws-codepipeline');
+import actions = require('@aws-cdk/aws-codepipeline-actions');
+import iam = require('@aws-cdk/aws-iam');
+import cdk = require('@aws-cdk/core');
+
+/**
+ * Simple three-stage pipeline to build the base image for the local container endpoints image.
+ * [GitHub source] -> [CodeBuild build, pushes image to DockerHub] -> [CodeBuild Verify, verifies the pushed images]
+ *
+ * TODO: use Docker manifest to create manifest for images under a common tag and ECR public
+ */
+class EcsLocalContainerEndpointsImagePipeline extends cdk.Stack {
+  constructor(parent: cdk.App, name: string, props?: cdk.StackProps) {
+    super(parent, name, props);
+
+    // Instantiate pipeline
+    const pipeline = new codepipeline.Pipeline(this, 'Pipeline', {
+      pipelineName: 'local-container-endpoints-image',
+    });
+
+    // Source stage
+    // Secret under ecs-local-container-endpoints+release@amazon.com
+    const githubAccessToken = cdk.SecretValue.secretsManager('EcsDevXGitHubToken');
+
+    const sourceOutput = new codepipeline.Artifact('SourceArtifact');
+    const sourceAction = new actions.GitHubSourceAction({
+      actionName: 'GitHubSource',
+      owner: 'awslabs',
+      repo: 'amazon-ecs-local-container-endpoints',
+      oauthToken: githubAccessToken,
+      branch: 'mainline',
+      output: sourceOutput
+    });
+
+    pipeline.addStage({
+      stageName: 'Source',
+      actions: [sourceAction],
+    });
+
+    // Build stage containing build project for each architecture image
+    const buildStage = pipeline.addStage({
+      stageName: 'Build',
+    });
+
+    const verifyStage = pipeline.addStage({
+      stageName: 'Verify',
+    });
+
+    const platforms = [
+      {'arch': 'amd64', 'buildImage': codebuild.LinuxBuildImage.AMAZON_LINUX_2_3},
+      {'arch': 'arm64', 'buildImage': codebuild.LinuxBuildImage.AMAZON_LINUX_2_ARM},
+    ];
+
+    // Create build and verify project for each platform
+    for (const platform of platforms) {
+      const arch = platform['arch'];
+
+      const buildProject = new codebuild.PipelineProject(this, `BuildImage-${arch}`, {
+        buildSpec: codebuild.BuildSpec.fromSourceFilename('./buildspec.yml'),
```

Comment:
nit: what do you think of moving the buildspec to the `infra` directory as well?

Reply:
So I actually did originally had it in the infra directory, but then moved it to the top level since that's where codebuild typically expects to find it and I thought it'd maybe be easier for it to find the Makefile that way. I also was following what [aws-for-fluent-bit](https://github.com/aws/aws-for-fluent-bit) was doing with where they were keeping their buildspec files, but maybe I was making incorrect assumptions about the codebuild file system. Do you think there's other advantages to putting it under the infra folder?

Reply:
nah it's fine to keep them at the root, i was just asking to keep the buildspec closer to the codebuild project declaration